parsing/dune

```diff
@@ -12,36 +12,20 @@
 ;*                                                                        *
 ;**************************************************************************
 
-;; We're just reusing the stuff from boot/ here.
```

Comment:
Here's an explanation:

I deleted the rules to generate `parser.ml` and `parser.mli` since they're now handled by dune's `menhir` rule.

I inlined the rules to generate `camlinternalMenhirLib.ml` and `camlinternalMenhirLib.mli` from the Makefile: they used to be generated in `boot/` by a `make` command and copied from there to here by dune, and now they're generated here instead.

Reply:
I think it's OK to take this more-or-less on faith, myself.